java/java.editor/src/org/netbeans/modules/java/editor/fold/JavaElementFoldManager.java

```diff
@@ -75,8 +75,8 @@ public class JavaElementFoldManager implements FoldManager {
     public static final FoldType JAVADOC_FOLD_TYPE = FoldType.DOCUMENTATION.derive("javadoc", Bundle.FoldType_Javadoc(), 
 	     new org.netbeans.api.editor.fold.FoldTemplate(3, 2, "/**...*/")); // NOI18N
 
-    @NbBundle.Messages("FoldType_Methods=Methods")
-    public static final FoldType CODE_BLOCK_FOLD_TYPE = FoldType.MEMBER.derive("method", Bundle.FoldType_Methods(), 
```

Comment:
Please do not **rename** an existing fold type -- this will break user settings. Instead, add an additional FoldType  - derive from nothing or FoldType.CODE_BLOCK.

Then adjust the code to use the new fold type for blocks, fors, ...

Such setup allows the user to control inner code folds and method-level folds separately, i.e. if one sets methods to be collapsed by default, the code blocks won't be initially collapsed when the user expands a method element.

Reply:
You are completly right, I thought so too. I will change this to Draft and will work on it further. Thx.

Reply:
Here is what I did:
* Fixed the visitBlock method and add check for ignoring method because visitMethod is already available
* Change code for visitMethod similar to visitClass but just for methodfolding
* Introduce new option for code block